packages/office-ui-fabric-react/src/utilities/DraggableZone/DraggableZone.tsx

```diff
@@ -0,0 +1,339 @@
+import * as React from 'react';
+import { IDraggableZoneProps, ICoordinates, IDragData } from './DraggableZone.types';
+import { getClassNames } from './DraggableZone.styles';
+import { BaseComponent } from '../../Utilities';
+
+export interface IDraggableZoneState {
+  isDragging: boolean;
+  position: ICoordinates;
+  lastPosition?: ICoordinates;
+}
+
+const eventMapping = {
+  touch: {
+    start: 'touchstart',
+    move: 'touchmove',
+    stop: 'touchend'
+  },
+  mouse: {
+    start: 'mousedown',
+    move: 'mousemove',
+    stop: 'mouseup'
+  }
+};
+
+// These are needed so that we can generalize the events
+// and so we have access to clientX and clientY in the touch events
+type MouseTouchEvent<T> = React.MouseEvent<T> & React.TouchEvent<T>;
+
+export class DraggableZone extends BaseComponent<IDraggableZoneProps, IDraggableZoneState> {
+  private _touchId?: number;
```

Comment:
Can you use React.Component or React.PureComponent? for eventing:

```tsx
import { on } from '../../Utilities';

  this._events = [
    on(document, eventName, cb),
    on(document, ...)
  ];
```
to unmount:

```tsx
this._events.forEach(dispose => dispose());
```

This ends up reducing bundle size of DraggableZone significantly. You can alternatively just have a private instance of eventgroup, but it's chunkier.

Reply:
Could I just make `this._events` be an `EventGroup` and then just use `on`/`off` like before?

---
In reply to: [278373175](https://github.com/OfficeDev/office-ui-fabric-react/pull/8426#discussion_r278373175) [](ancestors = 278373175)

Reply:
Do you need all the logic of `EventGroup`? If you just need `on` / `off` then the helper @dzearing proposed should be functionally equivalent.